src/gossip/service.zig

```diff
@@ -906,7 +907,7 @@ pub const GossipService = struct {
             defer gossip_table_lock.unlock();
 
             const gossip_table: *const GossipTable = gossip_table_lock.get();
-            break :blk gossip_table.getEntriesWithCursor(&buf, push_cursor);
+            break :blk gossip_table.getEntriesWithCursor(&buf, push_cursor); // MEM1: CLONE HERE; OR
```

Comment:
Yeah I left this one because I actually wanted to briefly discuss the best approach here. We can either move the logic responsible for building messages into the scope of the read lock, or clone the entries we pull out of the gossip table. This is an instance where we cant use thread safe contact info because we are serialising the heap allocated data when constructing the messages. I don't have a strong intuition over which approach would be better performance wise, so unless one of the two above options jumps out at you as the clear choice perhaps Ill need to figure out a way to benchmark